src/components/calcite-time-picker/calcite-time-picker.tsx

```diff
@@ -0,0 +1,702 @@
+    return (
+      <Host>
+        <div class="time-picker">
+          <div>
+            <span
+              aria-label="increase hour"
+              class="hour-up shift top-left"
+              onClick={this.incrementHour}
+              onKeyDown={this.incrementHour}
+              role="button"
+              tabIndex={-1}
+            >
+              <calcite-icon icon="chevronup" scale={iconScale} />
+            </span>
+            <span
+              aria-label="Hour"
+              aria-valuemax="23"
+              aria-valuemin="1"
+              aria-valuenow={this.hour !== "--" ? parseInt(this.hour) : undefined}
+              aria-valuetext={this.hour !== "--" ? this.hour : undefined}
+              class="hour input"
+              onBlur={this.hourBlurHandler}
+              onFocus={this.focusHandler}
+              onKeyDown={this.hourKeyDownHandler}
+              ref={(el) => (this.hourEl = el)}
+              role="spinbutton"
+              tabIndex={0}
+            >
+              {this.getDisplayHour()}
+            </span>
+            <span
+              aria-label="decrease hour"
+              class="hour-down shift bottom-left"
+              onClick={this.decrementHour}
+              onKeyDown={this.decrementHour}
+              role="button"
+              tabIndex={-1}
+            >
+              <calcite-icon icon="chevrondown" scale={iconScale} />
+            </span>
+          </div>
+          <span class="colon">:</span>
+          <div>
+            <span
+              aria-label="increase minute"
+              class="minute-up shift"
+              onClick={this.incrementMinute}
+              onKeyDown={this.incrementMinute}
+              role="button"
+              tabIndex={-1}
+            >
+              <calcite-icon icon="chevronup" scale={iconScale} />
+            </span>
+            <span
+              aria-label="Minute"
+              aria-valuemax="12"
+              aria-valuemin="1"
+              aria-valuenow={this.minute !== "--" ? parseInt(this.minute) : undefined}
+              aria-valuetext={this.minute !== "--" ? this.minute : undefined}
+              class="minute input"
+              onFocus={this.focusHandler}
+              onKeyDown={this.minuteKeyDownHandler}
+              ref={(el) => (this.minuteEl = el)}
+              role="spinbutton"
+              tabIndex={0}
+            >
+              {this.minute}
+            </span>
+            <span
+              aria-label="decrease minute"
+              class="minute-down shift"
+              onClick={this.decrementMinute}
+              onKeyDown={this.decrementMinute}
+              role="button"
+              tabIndex={-1}
+            >
+              <calcite-icon icon="chevrondown" scale={iconScale} />
+            </span>
+          </div>
+          {includeSeconds && <span class="colon">:</span>}
+          {includeSeconds && (
+            <div>
+              <span
```

Comment:
Can you use a button element here?

Reply:
I originally had this as a `<button>` but strangely enough in both Firefox and Safari there was this very odd bug that I couldn't figure out the root cause where clicking the button _didn't_ result in the button's click handler firing _before_ the `inputBlurHandler` function defined on the input in `input-time-picker` fired, which resulted in the popover being closed prematurely and never coming back.  For some super strange reason a `<span>` doesn't reproduce this behavior.  `<button>` works fine with Chromium browsers, just not Safari and Firefox for some reason.

Reply:
IMO it seems like we should use a button and do some more investigation.